### ISSUE TITLE
Style Dictionary crashes with generic error "Name must be a string" when there is a design token without a name

### ISSUE THREAD

**Issue by d49017** [amzn/style-dictionary#1102]
When there is a design token that happens to not have any name, so for example it looks like this `{ "": { "value": "#", "type": "color" } }`, the style dictionary crashes with a really generic error "Name must be a string" which is not helpful at all. I was only able to find this by removing tokens one-by-one.

It would be helpful to know a bit more about the problematic token, like for example where it is (JSONPath or at least line and column), "how it looks like" or what is the exact problem. Would it possible to improve error messages a bit?

node version: 18.14.2
npm version: 9.5.0
style-dictionary version: 3.9.2

**Comment by jorenbroekema** [comment]
I think we can improve some of the logs indeed by at least telling the user the filePath of the offending token so it's easier to find. If the token doesn't have a name, it's a bit tricky to tell the user which token has the issue though :\

**Comment by jorenbroekema** [comment]
True, we can use `token.path.join('.')`, PRs welcome! as of writing, use the v4 branch for now, if that branch no longer exists it means 4.0.0 has been released and the main branch can be used instead.
